pkg/logs/adapter/minio.go

```
@@ -0,0 +1,188 @@
+package adapter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/minio/minio-go/v7"
+	"go.uber.org/zap"
+
+	"github.com/kubeshop/testkube/pkg/log"
+	"github.com/kubeshop/testkube/pkg/logs/events"
+	minioconnecter "github.com/kubeshop/testkube/pkg/storage/minio"
+)
+
+const (
+	defaultBufferSize = 1024 * 100 // 100KB
+	defaultWriteSize  = 1024 * 80  // 80KB
+)
+
+var _ Adapter = &MinioConsumer{}
+
+type ErrMinioConsumerDisconnected struct {
+}
+
+func (e ErrMinioConsumerDisconnected) Error() string {
+	return "minio consumer disconnected"
+}
+
+type BufferInfo struct {
+	Buffer *bytes.Buffer
+	Part   int
+}
+
+// MinioConsumer creates new MinioSubscriber which will send data to local MinIO bucket
+func NewMinioConsumer(endpoint, accessKeyID, secretAccessKey, region, token, bucket string, opts ...minioconnecter.Option) *MinioConsumer {
+	c := &MinioConsumer{
+		minioConnecter: minioconnecter.NewConnecter(endpoint, accessKeyID, secretAccessKey, region, token, bucket, log.DefaultLogger, opts...),
+		Log:            log.DefaultLogger,
+		bucket:         bucket,
+		region:         region,
+		disconnected:   false,
+		buffInfos:      make(map[string]BufferInfo),
+	}
+	minioClient, err := c.minioConnecter.GetClient()
+	if err != nil {
+		c.Log.Errorw("error connecting to minio", "err", err)
```

so, we ignore the errors, only logging them?